### PR TITLE
first pass at using genuine entitlements endpoint

### PR DESCRIPTION
**Pull request by tavvy**
🐿 v2.7.0

### CODE REVIEW

File: src/client/swg-controller.js [hunk 1]
```diff
@@ -69,20 +69,31 @@ module.exports = class SwgController {
...
 						})
 						.catch(err => {
 							/* signal error */
 							events.signalError(err);
 							/* set onward journey */
 							this.handlers.onResolvedEntitlements({ promptLogin: true, entitlements: res.entitlements, error: err });
 						});
-				} else if (this.subscribeButtons) {
+				}
+				else if (res.hasEntitlements) {
```
Comment (remybach) [inline src/client/swg-controller.js]: Being pedantic, but can the `}` and `else if` be on the same line?

File: src/client/swg-controller.js [hunk 2]
```diff
@@ -69,20 +69,31 @@ module.exports = class SwgController {
...
+					/**
+					 * User has entitlements but not to requested content
+					 * TODO
+					 * - check if user has an active FT.com session
+					 * - prompt login if they do not
+					 * - UX message
+					 * - FUTURE (?) enable "upgrade" SwG buttons
+					 * */
+				}
+				else if (this.subscribeButtons) {
```
Comment (remybach) [inline src/client/swg-controller.js]: Ditto

File: src/client/swg-controller.js [hunk 3]
```diff
@@ -193,24 +202,37 @@ module.exports = class SwgController {
...
+	defaultOnwardEntitledJourney ({ promptLogin=false, consentRequired=false }={}) {
 		if (promptLogin) {
 			const loginHref = browser.generateLoginUrl();
 			this.overlay.show(`<p>It looks like you already have an FT.com subscription with Google.<br /><a href="${loginHref}">Please login</a><br /><br /><small>code: ENTITLED_LOGIN_REQUIRED</small></p>`);
 		} else {
 			const uuid = browser.getContentUuidFromUrl();
 			const contentHref = uuid ? `https://www.ft.com/content/${uuid}` : 'https://www.ft.com';
-			this.overlay.show(`<p>It looks like you already have an FT.com subscription with Google. You have been logged in.<br /><a href="${contentHref}">Go to content</a><br /><br /><small>code: ENTITLED_LOGIN_SUCCESS</small></p>`);
+			const consentHref = consentRequired && this.POST_SUBSCRIBE_URL + (uuid ? '&ft-content-uuid=' + uuid : '');
+			this.overlay.show(`<p>It looks like you already have an FT.com subscription with Google. You have been logged in.<br /><a href="${consentHref || contentHref}">Go to content</a><br /><br /><small>code: ENTITLED_LOGIN_SUCCESS</small></p>`);
```
Comment (remybach) [inline src/client/swg-controller.js]: Are we going to be keeping the code at the bottom of this overlay after release? Just wondering whether it's useful to show to the user at all.

Comment (tavvy) [reply src/client/swg-controller.js]: likely not. hopefully we wont be using our own overlay at all and invoking some swg message function.

**Pull request merged**